components/webui/server/src/plugins/MongoSocketIoServer/index.ts

```
@@ -92,6 +95,34 @@ class MongoSocketIoServer {
         const mongoDb = await initializeMongoClient(options);
+
+        // Fastify listens on all resolved addresses for localhost (e.g. `127.0.0.1`, `::1`), but
+        // socket.io only uses the main one. When multiple addresses are resolved, we proxy
+        // socket.io requests from secondary addresses to the main server.
+        try {
+            const addresses = await dnsLookup(fastify.config.HOST, {all: true});
+            if (1 < addresses.length) {
+                fastify.log.warn(
+                    `Multiple addresses found for ${fastify.config.HOST}: ${
+                        JSON.stringify(addresses)}. ` +
+                    "Proxying all socket.io requests to the main server."
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)